jsonnet/rules.jsonnet

```diff
@@ -219,6 +219,31 @@ local droppedKsmLabels = 'endpoint, instance, job, pod, service';
             |||,
             record: 'cluster:kube_persistentvolumeclaim_resource_requests_storage_bytes:provisioner:sum',
           },
+          {
+            // Track the number of physical cores that are considered accessible for general workloads to run on. A physical core is an unshared CPU as seen by the node operating system, ignoring hyperthreading or virtualization. The sum of all non-infrastructure node physical cores plus master node physical cores (if masters are schedulable) is considered available for workload use.
+            expr: 'sum(node_role_os_version_machine:cpu_capacity_cores:sum{label_node_role_kubernetes_io_master="",label_node_role_kubernetes_io_infra=""} or absent(node_role_os_version_machine:cpu_capacity_cores:sum{label_node_role_kubernetes_io_master="",label_node_role_kubernetes_io_infra=""})*0) + (sum(node_role_os_version_machine:cpu_capacity_cores:sum{label_node_role_kubernetes_io_master="true"}) * ((max(cluster_master_schedulable == 1)*0+1) or (absent(cluster_master_schedulable == 1)*0)))',
+            record: 'workload:capacity_physical_cpu_cores:sum',
+          },
+          {
+            // Record the rolling minimum of workload accessible physical CPU cores over a 5m window.
```

Comment:
if a spot instance is used less than 5m, its lost for billing, correct?  basically, we only bill now if the machine lives longer than 5m?

Reply:
Correct, we will use the max:5m metric to see whether this is happening and maybe change policy, but we'd start with the less restrictive story.